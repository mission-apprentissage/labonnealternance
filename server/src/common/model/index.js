import mongoosePaginate from "mongoose-paginate"
import { getElasticInstance, mongoosastic } from "../esClient/index.js"
import schema from "../model/schema/index.js"
import { mongooseInstance } from "../mongodb.js"

const createModel = (modelName, descriptor, options = {}) => {
  const schema = new mongooseInstance.Schema(descriptor)

  if (options.esIndexName) {
    schema.plugin(mongoosastic, { esClient: getElasticInstance(), index: options.esIndexName })
  }

  if (options.esIndexName || options.paginate) {
    schema.plugin(mongoosePaginate)
  }

  if (options.createMongoDBIndexes) {
    options.createMongoDBIndexes(schema)
  }

  return mongooseInstance.model(modelName, schema)
}

const DomainesMetiers = createModel("domainesmetiers", schema.domainesMetiersSchema, {
  esIndexName: "domainesmetiers",
})
const FormationCatalogue = createModel("formationcatalogues", schema.mnaFormationSchema, {
  esIndexName: "formationcatalogues",
})
const DiplomesMetiers = createModel("diplomesmetiers", schema.diplomesMetiersSchema, {
  esIndexName: "diplomesmetiers",
})
const ApiCalls = createModel("apicalls", schema.apiCallSchema)
const Application = createModel("applications", schema.applicationSchema, {
  paginate: true,
})
const SourceFormations = createModel("sourceformations", schema.sourceFormationsSchema)
const GeoLocation = createModel("geolocation", schema.geoLocationSchema)
const EmailBlacklist = createModel("emailblacklist", schema.emailBlacklist)
const Opco = createModel("opco", schema.opco)
const BonnesBoites = createModel("bonnesboites", schema.bonneBoiteSchema, {
  esIndexName: "bonnesboites",
})
const User = createModel("user", schema.userSchema, { paginate: true })
const Appointment = createModel("appointment", schema.appointmentSchema, { paginate: true })
const WidgetParameter = createModel("widgetParameter", schema.widgetParameterSchema, { paginate: true })
const Etablissement = createModel("etablissement", schema.etablissementSchema, { paginate: true })
const ParcoursupEtablissementStat = createModel("parcoursupEtablissementStat", schema.parcoursupEtablissementStatSchema, { paginate: true })
const Formulaire = createModel("formulaire", schema.formulaireSchema, { esIndexName: "formulaires", paginate: true })
const Credential = createModel("credentials", schema.credentialSchema)
const Offre = createModel("offre", schema.offresSchema)
const Optout = createModel("optout", schema.optoutSchema, { paginate: true })
const ReferentielOpco = createModel("referentielOpco", schema.referentielOpcoSchema)
const UserRecruteur = createModel("userRecruteur", schema.userRecruteurSchema, { paginate: true })
const AppointmentDetailed = createModel("appointmentDetailed", schema.appointmentSchema)

export {
  DomainesMetiers,
  FormationCatalogue,
  DiplomesMetiers,
  ApiCalls,
  Application,
  SourceFormations,
  GeoLocation,
  EmailBlacklist,
  Opco,
  BonnesBoites,
  User,
  Appointment,
  WidgetParameter,
  Etablissement,
  ParcoursupEtablissementStat,
  Formulaire,
  Credential,
  Offre,
  Optout,
  ReferentielOpco,
  UserRecruteur,
  AppointmentDetailed,
}
