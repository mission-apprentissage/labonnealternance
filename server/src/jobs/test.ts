import { ObjectId } from "bson"
import { IApplicant, IApplication, ZApplicant } from "shared"

import { getDbCollection } from "@/common/utils/mongodbUtils"

import { asyncForEach } from "../common/utils/asyncUtils"

type ApplicationAggregate = Pick<IApplication, "applicant_email" | "applicant_first_name" | "applicant_last_name" | "applicant_phone" | "_id">

export const migrationTest = async () => {
  // console.log(extensions.telephone.safeParse("0690760552"))
  // return
  const applications = (await getDbCollection("applications")
    .find({}, { projection: { applicant_first_name: 1, applicant_last_name: 1, applicant_email: 1, applicant_phone: 1, _id: 1 } })
    .toArray()) as ApplicationAggregate[]

  const stat = { error: 0, success: 0, total: applications.length }

  await asyncForEach(applications, async (application) => {
    const { applicant_first_name, applicant_email, applicant_last_name, applicant_phone, _id } = application
    const now = new Date()
    const applicant: IApplicant = {
      _id: new ObjectId(),
      firstname: applicant_first_name,
      lastname: applicant_last_name,
      phone: applicant_phone,
      email: applicant_email,
      last_connection: null,
      createdAt: now,
      updatedAt: now,
    }
    const validation = ZApplicant.safeParse(applicant)
    if (!validation.success) {
      stat.error++
      console.log({ applicant, err: validation.error.errors })
    } else {
      stat.success++
    }
  })
  console.log(stat)
}
