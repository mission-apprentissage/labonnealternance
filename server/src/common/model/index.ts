import { captureException } from "@sentry/node"

import { db, mongooseInstance } from "@/common/mongodb"

import { logger } from "../logger"

import AnonymizedUser from "./schema/anonymizedUsers/anonymizedUsers.schema"
import ApiCalls from "./schema/apiCall/apiCall.schema"
import AnonymizedApplication from "./schema/application/anonymizedApplications.schema"
import Application from "./schema/application/applications.schema"
import AppointmentDetailed from "./schema/appointmentDetailed/appointmentDetailed.schema"
import Appointment from "./schema/appointments/appointment.schema"
import Credential from "./schema/credentials/credential.schema"
import CustomEmailETFA from "./schema/customEmailETFA/customEmailETFA.schema"
import DiplomesMetiers from "./schema/diplomesmetiers/diplomesmetiers.schema"
import DomainesMetiers from "./schema/domainesmetiers/domainesmetiers.schema"
import EligibleTrainingsForAppointment from "./schema/eligibleTrainingsForAppointment/eligibleTrainingsForAppointment.schema"
import eligibleTrainingsForAppointmentHistory from "./schema/eligibleTrainingsForAppointmentsHistory/eligibleTrainingsForAppointmentHistory.schema"
import EmailBlacklist from "./schema/emailBlacklist/emailBlacklist.schema"
import Etablissement from "./schema/etablissements/etablissement.schema"
import FicheMetierRomeV4 from "./schema/ficheRomeV4/ficheRomeV4"
import FormationCatalogue from "./schema/formationCatalogue/formationCatalogue.schema"
import GeoLocation from "./schema/geolocation/geolocation.schema"
import InternalJobs from "./schema/internalJobs/internalJobs.schema"
import Job from "./schema/jobs/jobs.schema"
import LbaCompany from "./schema/lbaCompany/lbaCompany.schema"
import LbaCompanyLegacy from "./schema/lbaCompanylegacy/lbaCompanyLegacy.schema"
import { Cfa } from "./schema/multiCompte/cfa.schema"
import { Entreprise } from "./schema/multiCompte/entreprise.schema"
import { RoleManagement } from "./schema/multiCompte/roleManagement.schema"
import { UserWithAccount } from "./schema/multiCompte/userWithAccount.schema"
import Opco from "./schema/opco/opco.schema"
import Optout from "./schema/optout/optout.schema"
import Recruiter from "./schema/recruiter/recruiter.schema"
import RecruteurLbaUpdateEvent from "./schema/recruteurLbaUpdateEvent/recruteurLbaUpdateEvent.schema"
import ReferentielOnisep from "./schema/referentielOnisep/referentielOnisep.schema"
import ReferentielOpco from "./schema/referentielOpco/referentielOpco.schema"
import ReferentielRome from "./schema/referentielRome/referentielRome"
import Session from "./schema/session/session.schema"
import SiretDiffusibleStatus from "./schema/siretDiffusibleStatusSchema/siretDiffusibleStatusSchema.schema"
import UnsubscribedLbaCompany from "./schema/unsubscribedLbaCompany/unsubscribedLbaCompany.schema"
import UnsubscribeOF from "./schema/unsubscribedOF/unsubscribeOF.schema"
import User from "./schema/user/user.schema"

const createSpecialIndexes = async () => {
  await db.collection("bonnesboites").createIndex({ geopoint: "2dsphere" })
  await db.collection("formationcatalogues").createIndex({ lieu_formation_geopoint: "2dsphere" })
  await db.collection("recruiters").createIndex({ geopoint: "2dsphere" })
}

export async function createMongoDBIndexes() {
  const results = await Promise.allSettled(
    mongooseInstance.modelNames().map(async (name) => {
      const model = mongooseInstance.model(name)
      return model.createIndexes({ background: true }).catch(async (e) => {
        if (e.codeName === "IndexOptionsConflict" || e.codeName === "IndexKeySpecsConflict") {
          const err = new Error(`Conflict in indexes for ${name}`, { cause: e })
          logger.error(err)
          captureException(err)
          await mongooseInstance.connection.collection(model.collection.name).dropIndexes()
          await model.createIndexes({ background: true })
        } else {
          throw e
        }
      })
    })
  )

  await createSpecialIndexes()

  const errors = results.reduce((acc, r) => {
    if (r.status === "rejected") {
      acc.push(r.reason)

      logger.error(r.reason)
      captureException(r.reason)
    }

    return acc
  }, [] as Error[])

  if (errors.length > 0) {
    throw new AggregateError(errors, `createMongoDBIndexes failed with ${errors.length} errors`)
  }
}

export {
  AnonymizedApplication,
  AnonymizedUser,
  ApiCalls,
  Application,
  Appointment,
  AppointmentDetailed,
  Cfa,
  Credential,
  CustomEmailETFA,
  DiplomesMetiers,
  DomainesMetiers,
  EligibleTrainingsForAppointment,
  EmailBlacklist,
  Entreprise,
  Etablissement,
  FicheMetierRomeV4,
  FormationCatalogue,
  GeoLocation,
  InternalJobs,
  Job,
  LbaCompany,
  LbaCompanyLegacy,
  RecruteurLbaUpdateEvent,
  Opco,
  Optout,
  Recruiter,
  ReferentielOnisep,
  ReferentielOpco,
  ReferentielRome,
  RoleManagement,
  Session,
  SiretDiffusibleStatus,
  UnsubscribeOF,
  UnsubscribedLbaCompany,
  User,
  UserWithAccount,
  eligibleTrainingsForAppointmentHistory,
}
