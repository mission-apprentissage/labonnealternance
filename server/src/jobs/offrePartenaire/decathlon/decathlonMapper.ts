import { ObjectId } from "mongodb"
import { TRAINING_CONTRACT_TYPE } from "shared/constants/index"
import dayjs from "shared/helpers/dayjs"
import { JOBPARTNERS_LABEL, NIVEAUX_DIPLOMES_EUROPEENS } from "shared/models/jobsPartners.model"
import { IComputedJobsPartners } from "shared/models/jobsPartnersComputed.model"
import { z } from "zod"

import { blankComputedJobPartner } from "../fillComputedJobsPartners"

export const ZDecathlonJob = z
  .object({
    reference: z.string(),
    title: z.string(),
    description: z.string().nullish(),
    contract_type: z.string().nullish(),
    education_level: z.string().nullish(),
    skills: z.array(z.string()).nullish(),
    published_at: z.string().nullish(),
    contract_duration: z
      .object({
        min: z.string().nullish(),
      })
      .passthrough()
      .nullish(),
    brand: z
      .object({
        name: z.string().nullish(),
        description: z.string().nullish(),
      })
      .passthrough()
      .nullish(),
    entity: z
      .object({
        address: z
          .object({
            formatted: z.string().nullish(),
            position: z
              .object({
                lon: z.string().nullish(),
                lat: z.string().nullish(),
              })
              .passthrough()
              .nullish(),
            parts: z.object({
              street: z.string().nullish(),
              zip: z.string().nullish(),
              city: z.string().nullish(),
              county: z.string().nullish(),
              state: z.string().nullish(),
              country: z.string().nullish(),
            }),
          })
          .passthrough()
          .nullish(),
      })
      .passthrough()
      .nullish(),
    apply_url: z.string().nullish(),
  })
  .passthrough()

export type IDecathlonJob = z.output<typeof ZDecathlonJob>

export const decathlonJobToJobsPartners = (job: IDecathlonJob): IComputedJobsPartners => {
  const { contract_duration, reference, title, description, education_level, skills = [], published_at, brand, entity, apply_url, contract_type } = job
  const { address } = entity ?? {}
  const { position } = address ?? {}
  const { lat, lon } = position ?? {}

  const created_at = new Date()
  const partnerJob: IComputedJobsPartners = {
    ...blankComputedJobPartner(),
    _id: new ObjectId(),
    created_at,
    updated_at: created_at,
    partner_label: JOBPARTNERS_LABEL.DECATHLON,
    partner_job_id: reference,
    contract_type: contract_type === "Contrat d'alternance (pro, apprentissage)" ? [TRAINING_CONTRACT_TYPE.APPRENTISSAGE] : undefined,
    contract_remote: null,
    contract_duration: contract_duration?.min ? parseInt(contract_duration.min, 10) : null,
    offer_title: title,
    offer_description: description,
    offer_target_diploma: getDiplomaLevel(education_level),
    offer_desired_skills: skills,
    offer_access_conditions: [],
    offer_multicast: true,
    offer_to_be_acquired_skills: [],
    offer_rome_codes: undefined,
    offer_creation: published_at ? new Date(published_at) : null,
    offer_expiration: dayjs
      .tz(published_at || created_at)
      .add(2, "months")
      .toDate(),
    workplace_siret: null,
    workplace_name: brand?.name,
    workplace_description: brand?.description,
    workplace_address_label: address?.formatted,
    workplace_address_zipcode: address?.parts?.zip,
    workplace_address_city: address?.parts?.city,
    workplace_address_street_label: address?.parts?.street,
    workplace_geopoint:
      lat && lon
        ? {
            type: "Point",
            coordinates: [parseFloat(lon), parseFloat(lat)],
          }
        : undefined,
    apply_url: formatApplyUrl(apply_url),
  }
  return partnerJob
}

function getDiplomaLevel(value: string | null | undefined) {
  if (!value) return null
  const mapping = {
    "Aucun diplÃ´me, CEP ou BEPC": "3",
    "CAP, BEP": "3",
    "Bac, bac professionnel": "4",
    "BTS, Bac+2": "5",
    "BUT, Licence, Bac+3": "6",
    "Maitrise, IEP, IUP, Bac+4": "6",
    "Master, Bac+5": "6",
    "Doctorat, Bac+8": "6",
  }
  const diplomeValue = mapping[value]
  const found = NIVEAUX_DIPLOMES_EUROPEENS.find((x) => x.value === diplomeValue)
  if (!found) return null
  return {
    european: found.value,
    label: found.label,
  }
}

function formatApplyUrl(url: string | null | undefined): string | null {
  if (!url) return null
  const urlObj = new URL(url)
  urlObj.searchParams.append("s_o", "la-bonne-alternance")
  return urlObj.toString()
}
