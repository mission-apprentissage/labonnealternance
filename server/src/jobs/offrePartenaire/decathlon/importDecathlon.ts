import axios from "axios"
import { JOBPARTNERS_LABEL } from "shared/models/jobsPartners.model"
import rawDecathlonModel from "shared/models/rawDecathlon.model"
import z from "zod"

import config from "@/config"
import { importFromStreamInJson } from "@/jobs/offrePartenaire/importFromStreamInJson"

import { rawToComputedJobsPartners } from "../rawToComputedJobsPartners"

import { decathlonJobToJobsPartners, ZDecathlonJob } from "./decathlonMapper"

const rawCollectionName = rawDecathlonModel.collectionName

const ZJsonFile = z
  .object({
    ads: z.array(ZDecathlonJob),
  })
  .passthrough()

export const importDecathlonRaw = async (sourceStream?: NodeJS.ReadableStream) => {
  if (!sourceStream) {
    const response = await axios.get(config.decathlonUrl, {
      responseType: "stream",
    })
    sourceStream = response.data
  }
  await importFromStreamInJson({
    destinationCollection: rawCollectionName,
    stream: sourceStream!,
    partnerLabel: JOBPARTNERS_LABEL.HELLOWORK,
    getOffers(json) {
      const parsed = ZJsonFile.parse(json)
      return parsed.ads
    },
  })
}

export const importDecathlonToComputed = async () => {
  await rawToComputedJobsPartners({
    collectionSource: rawCollectionName,
    partnerLabel: JOBPARTNERS_LABEL.DECATHLON,
    zodInput: ZDecathlonJob,
    mapper: decathlonJobToJobsPartners,
  })
}
