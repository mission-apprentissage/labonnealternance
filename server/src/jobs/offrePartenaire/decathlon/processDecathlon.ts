import { importDecathlonRaw, importDecathlonToComputed } from "@/jobs/offrePartenaire/decathlon/importDecathlon"

export const processDecathlon = async () => {
  await importDecathlonRaw()
  await importDecathlonToComputed()
}
