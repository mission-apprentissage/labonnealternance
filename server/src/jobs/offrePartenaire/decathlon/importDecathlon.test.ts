import fs from "node:fs"

import { JOBPARTNERS_LABEL } from "shared/models/jobsPartners.model"
import { beforeEach, describe, expect, it, vi } from "vitest"

import { getDbCollection } from "@/common/utils/mongodbUtils"
import { importDecathlonRaw, importDecathlonToComputed } from "@/jobs/offrePartenaire/decathlon/importDecathlon"
import { useMongo } from "@tests/utils/mongo.test.utils"

const now = new Date("2024-07-21T04:49:06.000+02:00")

describe("importDecathlon", () => {
  useMongo()

  beforeEach(() => {
    vi.useFakeTimers()
    vi.setSystemTime(now)

    return async () => {
      vi.useRealTimers()
      await getDbCollection("computed_jobs_partners").deleteMany({})
      await getDbCollection("raw_decathlon").deleteMany({})
    }
  })

  it("should test the import of hellowork data into computed_job_partners", async () => {
    const fileStream = fs.createReadStream("server/src/jobs/offrePartenaire/decathlon/decathlon.test.input.json")
    await importDecathlonRaw(fileStream)
    expect.soft(await getDbCollection("raw_decathlon").countDocuments({})).toBe(3)

    await importDecathlonToComputed()
    const jobs = (
      await getDbCollection("computed_jobs_partners")
        .find({ partner_label: JOBPARTNERS_LABEL.DECATHLON }, { projection: { _id: 0, created_at: 0 } })
        .toArray()
    ).sort((a, b) => ((a.partner_job_id ?? "") < (b.partner_job_id ?? "") ? -1 : 1))
    expect.soft(jobs.length).toBe(3)
    expect.soft(jobs).toMatchSnapshot()
  })
})
