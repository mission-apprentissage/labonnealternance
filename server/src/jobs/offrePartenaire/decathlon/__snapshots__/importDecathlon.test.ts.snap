// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`importDecathlon > should test the import of hellowork data into computed_job_partners 1`] = `
[
  {
    "apply_phone": null,
    "apply_url": "https://joinus.decathlon.fr/fr/annonce/3975237-assistant-e-responsable-securite-exploitation-hse-hf-niveau-master-64700-hendaye?s_o=la-bonne-alternance#declareStep1",
    "business_error": null,
    "contract_duration": 12,
    "contract_is_disabled_elligible": false,
    "contract_remote": null,
    "contract_start": null,
    "contract_type": [
      "Apprentissage",
    ],
    "errors": [],
    "jobs_in_success": [],
    "offer_access_conditions": [],
    "offer_creation": 2025-09-23T12:07:38.000Z,
    "offer_description": "<div><p>TES FUTURES MISSIONS : </p><p>Aux côtés de notre responsable d'exploitation, tu garantis la sécurité du bâtiment, des biens et des personnes. En relation avec les prestataires, tu garantis la maintenance des installations des bâtiments à travers la gestion du planning des interventions. </p><br><p>Pour mettre en œuvre tes projets, tu assures le développement et le pilotage de groupes de travail. Ta mission sera d'intégrer le maximum de collaborateurs avec toi, dans les décisions et la mise en œuvre des plans d’actions. Tu accompagnes également les équipes en préparant les différents audits internes. </p><br><p>Garant-e de ton périmètre, tu t’assures du respect des réglementations sur le site. Tu travailles l’évolution des indicateurs de performance et recherche l'amélioration continue des procédures. Tu as aussi pour responsabilité le respect du budget attribué. </p><br><p>En bref, tu assures la sécurité de tes coéquipiers. Pour t’accompagner dans cette mission, tu pourras bien sûr toujours compter sur ton/ta tuteur/tutrice !</p></div>",
    "offer_desired_skills": [
      "curieu(se)",
      "autonome",
      "rigoureu(se)",
    ],
    "offer_expiration": 2025-11-23T11:07:38.000Z,
    "offer_multicast": true,
    "offer_opening_count": 1,
    "offer_origin": null,
    "offer_rome_codes": null,
    "offer_status": "Active",
    "offer_status_history": [],
    "offer_target_diploma": {
      "european": "6",
      "label": "Licence, Maîtrise, autres formations (Bac+3 à Bac+4)",
    },
    "offer_title": "Assistant.e Responsable Sécurité Exploitation HSE H/F - Niveau Master",
    "offer_to_be_acquired_knowledge": [],
    "offer_to_be_acquired_skills": [],
    "partner_job_id": "APEKPPZCVVXREV449",
    "partner_label": "Décathlon",
    "updated_at": 2024-07-21T02:49:06.000Z,
    "validated": false,
    "workplace_address_city": "Hendaye",
    "workplace_address_label": "8 Rue des Orangers,  64700 Hendaye, France",
    "workplace_address_street_label": "8 Rue des Orangers",
    "workplace_address_zipcode": "64700",
    "workplace_brand": null,
    "workplace_description": "DECATHLON Real Estate",
    "workplace_geopoint": {
      "coordinates": [
        -1.7799301,
        43.3689738,
      ],
      "type": "Point",
    },
    "workplace_idcc": null,
    "workplace_legal_name": null,
    "workplace_naf_code": null,
    "workplace_naf_label": null,
    "workplace_name": "DECATHLON Real Estate",
    "workplace_opco": null,
    "workplace_siret": null,
    "workplace_size": null,
    "workplace_website": null,
  },
  {
    "apply_phone": null,
    "apply_url": "https://joinus.decathlon.fr/fr/annonce/3977401-alternance-manager-commerce-hf-78140-velizy-villacoublay?s_o=la-bonne-alternance#declareStep1",
    "business_error": null,
    "contract_duration": 12,
    "contract_is_disabled_elligible": false,
    "contract_remote": null,
    "contract_start": null,
    "contract_type": [
      "Apprentissage",
    ],
    "errors": [],
    "jobs_in_success": [],
    "offer_access_conditions": [],
    "offer_creation": 2025-09-24T10:18:17.000Z,
    "offer_description": "<p>Chez Decathlon l’<b>alternance</b> est au cœur de nos préoccupations. Riche de plus de 300 métiers regroupés dans 8 filières, Decathlon te permettra d’acquérir un grand nombre de compétences humaines et professionnelles dans le secteur qui t’attire. Si tu es étudiant(e), passionné(e) de sport, que tu aimes les challenges et que tu souhaites te professionnaliser en parallèle de ton parcours scolaire…<b> DECATHLON est sans hésiter le “terrain de jeu”  idéal ! </b></p>

<p style="text-align:center;"><b>#CHOISIS TA TEAM# </b></p>

<p>Une stratégie qui contribue au parcours de milliers de jeunes en France et qui nous permettra de recruter nos talents sportifs de demain. Une politique qui fera de Decathlon un acteur majeur dans l'insertion professionnelle et sociale des jeunes en France<br><b>DECATHLON = une opportunité vers mon 1er emploi</b></p>

<p><b>Une ambition forte : 100% de nos alternants satisfaits de leur expérience chez Decathlon.</b><br>Decathlon considère ses alternants comme des coéquipiers CDI.<br></p>

<p><b>Nous portons un sens fort : Être utile aux gens et à leur planète. </b></p>

<p>Le sport n'a de sens que dans un environnement préservé. Nous développons nos activités sans dégrader nos terrains de jeux en proposant des services liés à l'économie circulaire (Occasion / location /réparation / recyclage) en proposant des produits éco-conçus en sourçant des énergies renouvelables...<br></p>

<p>Decathlon est engagé dans l'inclusion et la non-discrimination, et agit ainsi quotidiennement en faveur du handicap, des séniors, de la mixité sociale, de l'égalité entre les femmes et les hommes. <b>Chez Decathlon, nous recrutons avant tout des personnalités. </b><br></p>

<p><b>Rejoins DECATHLON et intègre une entreprise où il fait bon vivre<br></b>Viens partager une passion collective pour le sport avec une équipe unie et motivée. Marque des points en prenant des responsabilités, exprime ton sens de l’initiative et contribue à <b>rendre durablement le sport au plus grand nombre</b>. Une diversité de parcours s’offre à toi dans un environnement dynamique et challengeant. Dès ton arrivée, tu prends la responsabilité d’une équipe au sein d’un magasin.</p>

<p><b>Ta mission met en valeur ton leadership<br></b>Entre dans le jeu avec la responsabilité d’un <b>projet Humain et Commercial</b>.<br>Tu définis et mets en œuvre ce projet avec l’équipe que tu animes. Ensemble, vous relevez les défis que vous vous êtes fixés.</p>

<p><b>Ton terrain de jeu te permet de mettre tes compétences au service du sport<br></b>Tu es le·a coach·e/Leader de l’équipe, tu as la responsabilité  managériale de celle-ci : le recrutement, la gestion des parcours, la formation et le développement des compétences, la planification.</p>

<p>Plus important encore, tu assures le bien-être de chaque co-équipier-e.</p>

<p>Tu définis et tu animes le projet commercial de ton rayon : gestion de la gamme de tes produits, du  linéaire, des stocks, et du dynamisme de vente. Tu joues collectif : toute l’équipe est ambassadrice de nos produits !</p>

<p>Responsable de ton compte d'exploitation et véritable pilote de ton activité, tu développes et gères tes ressources en toute autonomie pour atteindre les objectifs. Tu fais équipe avec le Leader du magasin qui te manage, et tu bénéficies d’un plan de formation pour t’accompagner dans la réussite de l’ensemble de tes missions.</p>",
    "offer_desired_skills": [
      "entrepreneur",
      "Esprit d'équipe",
      "Management",
      "Accueil",
      "Produit",
      "Relation Sportif",
    ],
    "offer_expiration": 2025-11-24T09:18:17.000Z,
    "offer_multicast": true,
    "offer_opening_count": 1,
    "offer_origin": null,
    "offer_rome_codes": null,
    "offer_status": "Active",
    "offer_status_history": [],
    "offer_target_diploma": {
      "european": "6",
      "label": "Licence, Maîtrise, autres formations (Bac+3 à Bac+4)",
    },
    "offer_title": "ALTERNANCE Manager Commerce (H/F)",
    "offer_to_be_acquired_knowledge": [],
    "offer_to_be_acquired_skills": [],
    "partner_job_id": "KWYLVLYSRRKVGO513",
    "partner_label": "Décathlon",
    "updated_at": 2024-07-21T02:49:06.000Z,
    "validated": false,
    "workplace_address_city": "Vélizy-Villacoublay",
    "workplace_address_label": "2 Avenue de l'Europe,  78140 Vélizy-Villacoublay, France",
    "workplace_address_street_label": "2 Avenue de l'Europe",
    "workplace_address_zipcode": "78140",
    "workplace_brand": null,
    "workplace_description": "DECATHLON Retail Omnichannel",
    "workplace_geopoint": {
      "coordinates": [
        2.2201465,
        48.7812805,
      ],
      "type": "Point",
    },
    "workplace_idcc": null,
    "workplace_legal_name": null,
    "workplace_naf_code": null,
    "workplace_naf_label": null,
    "workplace_name": "DECATHLON Retail Omnichannel",
    "workplace_opco": null,
    "workplace_siret": null,
    "workplace_size": null,
    "workplace_website": null,
  },
  {
    "apply_phone": null,
    "apply_url": "https://joinus.decathlon.fr/fr/annonce/3973345-alternance-manager-commerce-hf-19100-brive-la-gaillarde?s_o=la-bonne-alternance#declareStep1",
    "business_error": null,
    "contract_duration": 12,
    "contract_is_disabled_elligible": false,
    "contract_remote": null,
    "contract_start": null,
    "contract_type": [
      "Apprentissage",
    ],
    "errors": [],
    "jobs_in_success": [],
    "offer_access_conditions": [],
    "offer_creation": 2025-09-22T15:32:41.000Z,
    "offer_description": "<p>Chez Decathlon l’<b>alternance</b> est au cœur de nos préoccupations. Riche de plus de 300 métiers regroupés dans 8 filières, Decathlon te permettra d’acquérir un grand nombre de compétences humaines et professionnelles dans le secteur qui t’attire. Si tu es étudiant(e), passionné(e) de sport, que tu aimes les challenges et que tu souhaites te professionnaliser en parallèle de ton parcours scolaire…<b> DECATHLON est sans hésiter le “terrain de jeu”  idéal ! </b></p>

<p style="text-align:center;"><b>#CHOISIS TA TEAM# </b></p>

<p>Une stratégie qui contribue au parcours de milliers de jeunes en France et qui nous permettra de recruter nos talents sportifs de demain. Une politique qui fera de Decathlon un acteur majeur dans l'insertion professionnelle et sociale des jeunes en France<br><b>DECATHLON = une opportunité vers mon 1er emploi</b></p>

<p><b>Une ambition forte : 100% de nos alternants satisfaits de leur expérience chez Decathlon.</b><br>Decathlon considère ses alternants comme des coéquipiers CDI.<br></p>

<p><b>Nous portons un sens fort : Être utile aux gens et à leur planète. </b></p>

<p>Le sport n'a de sens que dans un environnement préservé. Nous développons nos activités sans dégrader nos terrains de jeux en proposant des services liés à l'économie circulaire (Occasion / location /réparation / recyclage) en proposant des produits éco-conçus en sourçant des énergies renouvelables...<br></p>

<p>Decathlon est engagé dans l'inclusion et la non-discrimination, et agit ainsi quotidiennement en faveur du handicap, des séniors, de la mixité sociale, de l'égalité entre les femmes et les hommes. <b>Chez Decathlon, nous recrutons avant tout des personnalités. </b><br></p>

<p><b>Rejoins DECATHLON et intègre une entreprise où il fait bon vivre<br></b>Viens partager une passion collective pour le sport avec une équipe unie et motivée. Marque des points en prenant des responsabilités, exprime ton sens de l’initiative et contribue à <b>rendre durablement le sport au plus grand nombre</b>. Une diversité de parcours s’offre à toi dans un environnement dynamique et challengeant. Dès ton arrivée, tu prends la responsabilité d’une équipe au sein d’un magasin.</p>

<p><b>Ta mission met en valeur ton leadership<br></b>Entre dans le jeu avec la responsabilité d’un <b>projet Humain et Commercial</b>.<br>Tu définis et mets en œuvre ce projet avec l’équipe que tu animes. Ensemble, vous relevez les défis que vous vous êtes fixés.</p>

<p><b>Ton terrain de jeu te permet de mettre tes compétences au service du sport<br></b>Tu es le·a coach·e/Leader de l’équipe, tu as la responsabilité  managériale de celle-ci : le recrutement, la gestion des parcours, la formation et le développement des compétences, la planification.</p>

<p>Plus important encore, tu assures le bien-être de chaque co-équipier-e.</p>

<p>Tu définis et tu animes le projet commercial de ton rayon : gestion de la gamme de tes produits, du  linéaire, des stocks, et du dynamisme de vente. Tu joues collectif : toute l’équipe est ambassadrice de nos produits !</p>

<p>Responsable de ton compte d'exploitation et véritable pilote de ton activité, tu développes et gères tes ressources en toute autonomie pour atteindre les objectifs. Tu fais équipe avec le Leader du magasin qui te manage, et tu bénéficies d’un plan de formation pour t’accompagner dans la réussite de l’ensemble de tes missions.</p>",
    "offer_desired_skills": [
      "entrepreneur",
      "Esprit d'équipe",
      "Management",
      "Accueil",
      "Produit",
      "Relation Sportif",
    ],
    "offer_expiration": 2025-11-22T14:32:41.000Z,
    "offer_multicast": true,
    "offer_opening_count": 1,
    "offer_origin": null,
    "offer_rome_codes": null,
    "offer_status": "Active",
    "offer_status_history": [],
    "offer_target_diploma": {
      "european": "6",
      "label": "Licence, Maîtrise, autres formations (Bac+3 à Bac+4)",
    },
    "offer_title": "ALTERNANCE Manager Commerce (H/F)",
    "offer_to_be_acquired_knowledge": [],
    "offer_to_be_acquired_skills": [],
    "partner_job_id": "MAZEYQGIWWPKWM534",
    "partner_label": "Décathlon",
    "updated_at": 2024-07-21T02:49:06.000Z,
    "validated": false,
    "workplace_address_city": "Brive-la-Gaillarde",
    "workplace_address_label": "Le Mazaud,  19100 Brive-la-Gaillarde, France",
    "workplace_address_street_label": "Le Mazaud",
    "workplace_address_zipcode": "19100",
    "workplace_brand": null,
    "workplace_description": "DECATHLON Retail Omnichannel",
    "workplace_geopoint": {
      "coordinates": [
        1.477267,
        45.143855,
      ],
      "type": "Point",
    },
    "workplace_idcc": null,
    "workplace_legal_name": null,
    "workplace_naf_code": null,
    "workplace_naf_label": null,
    "workplace_name": "DECATHLON Retail Omnichannel",
    "workplace_opco": null,
    "workplace_siret": null,
    "workplace_size": null,
    "workplace_website": null,
  },
]
`;
